autoload/db/adapter/bigquery.vim

```diff
@@ -0,0 +1,76 @@
+let g:bq_global_ops = [
+  \ 'api',
+  \ 'api_version',
+  \ 'apilog',
+  \ 'bigqueryrc',
+  \ 'ca_certificates_file',
+  \ 'dataset_id',
+  \ 'debug_mode',
+  \ 'nodebug_mode',
+  \ 'disable_ssl_validation',
+  \ 'nodisable_ssl_validation',
+  \ 'discovery_file',
+  \ 'enable_gdrive',
+  \ 'noenable_gdrive',
+  \ 'fingerprint_job_id',
+  \ 'nofingerprint_job_id',
+  \ 'format',
+  \ 'headless',
+  \ 'noheadless',
+  \ 'httplib2_debuglevel',
+  \ 'job_id',
+  \ 'job_property',
+  \ 'jobs_query_use_request_id',
+  \ 'nojobs_query_use_request_id',
+  \ 'jobs_query_use_results_from_response',
+  \ 'nojobs_query_use_results_from_response',
+  \ 'location',
+  \ 'max_rows_per_request',
+  \ 'mtls',
+  \ 'nomtls',
+  \ 'project_id',
+  \ 'proxy_address',
+  \ 'proxy_password',
+  \ 'proxy_port',
+  \ 'proxy_username',
+  \ 'quiet',
+  \ 'noquiet',
+  \ 'synchronous_mode',
+  \ 'nosynchronous_mode',
+  \ 'trace',
+  \ 'use_regional_endpoints',
+  \ 'nouse_regional_endpoints',
+  \ ]
+
+function! s:command_for_url(params, action) abort
+  let cmd = ['bq']
+  let subcmd = [a:action]
+  for [k, v] in items(a:params)
+    let op = '--'.k.'='.v
+    if index(g:bq_global_ops, k) >= 0
+      call add(cmd, op)
+    else
+      call add(subcmd, op)
+    endif
+  endfor
+  return cmd + subcmd
+endfunction
+
+function! s:params(url) abort
+  let parsed_params = db#url#parse(a:url)
+  let conn_params = parsed_params.params
+  return conn_params
+endfunction
+
+function! db#adapter#bigquery#interactive(url, action) abort
+  return s:command_for_url(s:params(a:url), a:action)
+endfunction
+
+function! db#adapter#bigquery#input(url, in) abort
+  let out = []
+  if len(matchstr(a:in, '.sql'))
+    let out += db#adapter#bigquery#interactive(a:url, 'query')
+    let out += ['--flagfile='.a:in]
+  endif
+  return out
```

Comment:
What is the purpose of the `len(matchstr(a:in, '.sql'))` check? If it fails, the whole function will return `[]`, which will break things.

Reply:
## Before that code:

![image](https://user-images.githubusercontent.com/6231413/205487665-a50aac27-362a-49ef-8713-8bf77d763e4a.png)

as `bq` fails if there is no query.

## After

![image](https://user-images.githubusercontent.com/6231413/205487534-49fbbb9f-aa09-417f-8cd9-29f0b9161486.png)

What I see is at the begining `db#adapter#bigquery#input` is called with the url and a file called `0` and the actual query use a file named `1.sql` . I don't know the better way to deal with this so I decided to use only files with the extension `.sql`.

and after that the adapter works as expected, what I've checked:

```
:DB bigquery://?project_id=siad-1141&format=sparse&use_legacy_sql=false select 1
:%DB bigquery://?project_id=siad-1141&format=sparse&use_legacy_sql=false 
:'<,'>DB bigquery://?project_id=siad-1141&format=sparse&use_legacy_sql=false 
```

Reply:
That `0` file is it doing an auth check. What happens if you pass it to `--flagfile` like an SQL file?

Reply:
it continues running endlessly.

```
$ touch 0
$  bq --project_id='siad-1141' query --format=sparse --use_legacy_sql=false --flagfile=./0
```

![image](https://user-images.githubusercontent.com/6231413/205498264-0de0b66b-ec52-42cc-97ec-612bafaf3361.png)

Reply:
> Is that because of the file name, or is it because the file is empty? 
it's because it's empty. 

> if you close stdin with < /dev/null, does it still hapen?

here's the output 

![image](https://user-images.githubusercontent.com/6231413/205507064-c3c3da0b-9fa0-441c-80c5-5ae5a27813f2.png)

Reply:
Remember this?

> The purpose is to give the filename directly to the database, so that error messages, etc, can reference. Five adapters use this; the others accept input on stdin. Does bq really support neither?

And this?

> Either it needs to read from stdin or a file, or it's not becoming an official adapter.

I have repeatedly brought up sending the query on stdin. And now we have extremely solid evidence that if you don't give it a query, it tries to read one from stdin.

Please show me a version without `db#adapter#bigquery#input()`. It's clearly causing more problems than it solves.